tasks/eva_2950/split_rs_with_inconsistent_ss.py

```diff
@@ -0,0 +1,320 @@
+    try:
+        tree = ast.parse(astr)
+    except SyntaxError:
+        raise ValueError(astr)
+    for node in ast.walk(tree):
+        if isinstance(node, (ast.Module, ast.Expr, ast.Dict, ast.Str, ast.List, ast.Constant,
+                            ast.Attribute, ast.Num, ast.Name, ast.Load, ast.Tuple)):
+            continue
+        if isinstance(node, ast.Call) and isinstance(node.func, ast.Attribute) and node.func.attr == 'datetime':
+            continue
+        raise ValueError(astr)
+    return eval(astr)
+
+
+def parse_eva2850_diagnostic_log(log_file):
+    rsid = None
+    with open(log_file) as open_file:
+        for line in open_file:
+            line = line.strip()
+            if rsid:
+                # We are parsing the line just after finding the RS id
+                list_of_ssids = ast_parse_with_datetime(line.strip()[11:])
+                yield rsid, list_of_ssids
+                rsid = None
+            if 'Not all original SS has same info' in line:
+                sp_line = line.split()
+                rsid = int(sp_line[4].strip(','))
+
+        if rsid:
+            list_of_ssids = ast_parse_with_datetime(line.strip()[11:])
+            yield rsid, list_of_ssids
+
+
+genome_cache = {}
+
+
+def get_genome_object(ref_genome_directory, genome_accession):
+    if genome_accession not in genome_cache:
+        path_to_search = os.path.join(ref_genome_directory, '*', genome_accession, genome_accession + '.fa')
+        genome_paths = glob.glob(path_to_search)
+        if len(genome_paths) == 0:
+            raise ValueError(f'Cannot locate genome for {genome_accession} in {ref_genome_directory}')
+        genome_path = genome_paths[0]
+        genome_cache[genome_accession] = Fasta(genome_path, as_raw=True, read_ahead=40000)
+    return genome_cache[genome_accession]
+
+
+def variant_type(ref, alt):
+    if len(ref) == len(alt) == 1:
+        return 'SNP'
+    if len(ref) > 1 and len(alt) > 1:
+        return 'MNP'
+    if len(ref) > 1:
+        return 'DEL'
+    return 'INS'
+
+
+def grouper(iterable, n, fillvalue=None):
+    args = [iter(iterable)] * n
+    return zip_longest(fillvalue=fillvalue, *args)
+
+
+def shelve_submitted_variant_entities(mongo_handle, submitted_variant_ids):
+    output_collection = 'eva2950_dbsnpSubmittedVariantEntity'
+    batch_size = 1000
+    for batch_sve_ids in grouper(submitted_variant_ids, batch_size):
+        query_filter = {'_id': {'$in': batch_sve_ids}}
+        documents = [d for d in mongo_handle["eva_accession_sharded"]['dbsnpSubmittedVariantEntity'].find(query_filter)]
+        mongo_handle["eva_accession_sharded"][output_collection].insert(documents)
+        nb_sve = mongo_handle["eva_accession_sharded"][output_collection].estimated_document_count()
+        assert nb_sve == len(batch_sve_ids), 'Not all variants were transfer to the output collection ' + output_collection
+        response = mongo_handle['dbsnpSubmittedVariantEntity'].delete_many(query_filter)
```

Comment:
As we discussed some days back, we can just shelve the SS for now and do a delete after all the other shelving has been carried out in EVA-3004.